sympy/solvers/diophantine/tests/test_diophantine.py

```diff
@@ -660,11 +660,26 @@ def test_sum_of_three_squares():
               800, 801, 802, 803, 804, 805, 806]:
         a, b, c = sum_of_three_squares(i)
         assert a**2 + b**2 + c**2 == i
+        assert [a, b, c] == sorted([a, b, c])
+        assert a >= 0
+
+    # error
+    raises(ValueError, lambda: sum_of_three_squares(-1))
 
     assert sum_of_three_squares(7) is None
     assert sum_of_three_squares((4**5)*15) is None
-    assert sum_of_three_squares(25) == (5, 0, 0)
+    # if there are two zeros, there might be a solution
+    # with only one zero, e.g. 25 = (0, 3, 4)
+    assert sum_of_three_squares(25) == (0, 0, 5)
+    assert sum_of_three_squares(49) == (2, 3, 6)
     assert sum_of_three_squares(4) == (0, 0, 2)
+    nonzero = 0
+    for n in range(1000):
+        result = sum_of_three_squares(n)
+        if result:
+            nonzero += int(0 not in result)
+    assert nonzero == 786
```

Comment:
Should be 799. Only the following 201 numbers less than `10**3` cannot be represented non-trivially with 3 squares:
```
[0, 1, 2, 4, 5, 7, 8, 10, 13, 15, 16, 20, 23, 25, 28, 31, 32, 37, 39, 40, 47, 52, 55, 58, 60, 63, 64, 71, 79, 80, 85, 87, 92, 95, 100, 103, 111, 112, 119, 124, 127, 128, 130, 135, 143, 148, 151, 156, 159, 160, 167, 175, 183, 188, 191, 199, 207, 208, 215, 220, 223, 231, 232, 239, 240, 247, 252, 255, 256, 263, 271, 279, 284, 287, 295, 303, 311, 316, 319, 320, 327, 335, 340, 343, 348, 351, 359, 367, 368, 375, 380, 383, 391, 399, 400, 407, 412, 415, 423, 431, 439, 444, 447, 448, 455, 463, 471, 476, 479, 487, 495, 496, 503, 508, 511, 512, 519, 520, 527, 535, 540, 543, 551, 559, 567, 572, 575, 583, 591, 592, 599, 604, 607, 615, 623, 624, 631, 636, 639, 640, 647, 655, 663, 668, 671, 679, 687, 695, 700, 703, 711, 719, 727, 732, 735, 743, 751, 752, 759, 764, 767, 775, 783, 791, 796, 799, 807, 815, 823, 828, 831, 832, 839, 847, 855, 860, 863, 871, 879, 880, 887, 892, 895, 903, 911, 919, 924, 927, 928, 935, 943, 951, 956, 959, 960, 967, 975, 983, 988, 991, 999]
```